raven/processes/wps_hydrobasins_shape_selection.py

```diff
@@ -0,0 +1,158 @@
+import json
+import logging
+import re
+import tempfile
+
+import fiona as fio
+import geopandas as gpd
+from fiona.crs import from_epsg
+from pywps import LiteralInput, ComplexInput, ComplexOutput
+from pywps import Process, FORMATS
+from raven.utils import archive_sniffer, crs_sniffer
+from shapely.geometry import shape, Point
+
+LOGGER = logging.getLogger("PYWPS")
+
+
+class ShapeSelectionProcess(Process):
+    """Given lat/lon coordinates and a file containing vector data, return the feature containing the coordinates."""
+
+    def __init__(self):
+        inputs = [
+            LiteralInput('lonlat_coordinate', '(Lon, Lat) tuple for point of interest',
+                         data_type='string',
+                         default='(-68.724444, 50.646667)'),
+            LiteralInput('collect_upstream',
+                         'Attempt to capture both containing basin and all upstream basins from point',
+                         data_type='boolean',
+                         default='false'),
+            LiteralInput('crs', 'Coordinate Reference System of shape (EPSG) and lat/lon coordinates',
+                         data_type='integer',
+                         default=4326),
+            ComplexInput('shape', 'Vector Shape',
```

Comment:
No. The logic behind this process is specifically written for the structure of the HydroBASINS data set. I haven't come across anything that resembles the 'NEXT DOWN' identifier in any other products.

Reply:
So it makes no sense to leave that as a user parameter, correct ?

Reply:
True. My motivation for allowing someone to change that inputs is because the HydroBASINS data is available for multiple levels and with/without lakes. Should I just replace the input to include those as hard-coded options?

Reply:
You mean level = {10,11,12} and lake = {true, false} ? That would be my suggestion, yes.